GPSlog_Basic/app/src/main/java/com/example/gpslog/Segment.java

```diff
@@ -53,128 +63,149 @@ public Segment(ArrayList<Track> tracks){
     // Throws a ParseException if the SimpleDateFormat can not parse the time in the Tracks
     public void setFeatures(ArrayList<Track> tracks) throws ParseException {
 
-        setSegTimeLength(tracks);
-        setDistanceTraveled(tracks);
-
+        double currAcc = 0.0;
+        double currSpd = 0.0;
+        double accSum = 0.0;
+        int beginStopIndex = 0;
+        long currentStopTime;
         boolean isStopped = false;
         boolean hasStopped = false;
-        double maxPeek = 0.0;
-        Date beginningStopTime = new Date();
+        Track currTrack = tracks.get(0);
+        Track prevTrack;
+        double speedsSum = currTrack.speed;
+        maxSpeed = currTrack.speed;
+
+        //Handling the case where the app is started standing still (stop first)
+        if (currTrack.hiddenState == HMMClassifier.STOPPED) {
+            isStopped = true;
+            hasStopped = true;
+            beginStopIndex = 0;
+            numStops++;
+        }
 
-        // TODO: Make sure that this loop sets everything as required
         for (int i = 1; i < tracks.size(); i++) {
-            Track currTrack = tracks.get(i);
-            int currHiddenState = currTrack.hiddenState;
-            int prevHiddenState = tracks.get(i - 1).hiddenState;
-
+            prevTrack = currTrack;
+            currTrack = tracks.get(i);
             if (currTrack.speed > maxSpeed) {
                 maxSpeed = currTrack.speed;
             }
 
-            if(currHiddenState == HMMClassifier.STOPPED){
-                peakSpeed = maxPeek;
+            //Car comes to a stop
```

Comment:
If you want to break it into the four possible cases, I would suggest using a switch statement. Then the catch can be an Exception because the params were wrong

Reply:
Not sure about the best way to handle this, is there a way to pass multiple arguments to a switch statement? Or to do boolean logic in the code" case (boolean): "

eg. case (isStopped && status == Acceleration): doThis(); break;
eg. switch(isStopped, isAccelerating) { case etc: }

I could set up a variable that stores a string of the car's status beforehand but then that might just make the code more complicated.